components/skus/browser/rs/lib/src/sdk/credentials/mod.rs

```diff
@@ -24,80 +24,80 @@ where
         }
         Ok(())
     }
-
     #[instrument]
     pub async fn matching_order_credential_summary(
         &self,
         order_id: &str,
         domain: &str,
     ) -> Result<Option<CredentialSummary>, SkusError> {
-        let order = self.client.get_order(order_id).await?;
-
-        if let Some(order) = order {
-            let expires_at = order.expires_at;
-
-            if !order.location_matches(&self.environment, domain) {
-                return Err(InternalError::OrderLocationMismatch.into());
-            }
+        let wrapped_order = self.client.get_order(order_id).await?;
+        if wrapped_order.is_none() {
+          return Err(InternalError::NotFound.into());
+        }
+        let order = wrapped_order.unwrap();
+        if !order.location_matches(&self.environment, domain) {
+          return Err(InternalError::OrderLocationMismatch.into());
+        }
 
-            // FIXME only returns summary for first item
-            if let Some(item) = order.items.iter().next() {
-                return Ok(match item.credential_type {
-                    CredentialType::SingleUse => self
-                        .client
-                        .get_single_use_item_creds(&item.id)
+        let expires_at = order.expires_at;
+        for item in &order.items {
+          match item.credential_type {
+              CredentialType::SingleUse => {
+                let wrapped_creds = self
+                    .client
+                    .get_single_use_item_creds(&item.id)
+                    .await?;
+                if wrapped_creds.is_none() {
+                  continue;
+                }
+                let creds = wrapped_creds.unwrap();
```

Comment:
Same thing with `.unwrap()` here. If you can get away with `Result` and `Option` chain methods that's ideal; otherwise I'd recommend at least this:

```rust
if let Some(creds) = wrapped_creds {
    // ...
} else {
    continue;
}
```